internals/cli/error_handler.go

```diff
@@ -0,0 +1,33 @@
+package cli
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+func (c *CommandClause) argumentError(args []string) error {
+	if len(args) >= getRequired(c.Args) && len(args) <= len(c.Args) {
+		return nil
+	}
+	errorText, minimum, maximum := "", getRequired(c.Args), len(c.Args)
+
+	if strings.Contains(reflect.TypeOf(c.Args[0].Value).String(), "List") {
+		errorText += fmt.Sprintf(`"%s" requires at least %d %s.`, c.fullCommand(), minimum, pluralize("argument", minimum))
```

Comment:
We're using a side-effect (whether the type is named  here to determine if we're dealing with a list of arguments. Can't we just store that when we register it, so that we don't break this functionality if we ever rename a type?

Reply:
I found an alternative to solve this. Not sure it is the best approach. Let me know what you think.